servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -308,6 +316,11 @@ public StreamingHttpClient buildStreaming() {
                         ctx.builder.hostHeaderFilterFactoryFunction.apply(ctx.builder.address));
             }
 
+            if (ctx.builder.supportedEncodings != null) {
+                currClientFilterFactory = appendFilter(currClientFilterFactory,
```

Comment:
Can you please clarify why this filter has to be the last filter for the client (and the first for the server)?
I worry that if we do decompression first, users have to way to intercept the original payload body size and metrics filters will show only decompressed size.
We may think about an alternative approach for metrics if this is a strong requirement. Otherwise, it's better to expose just a filter and let users configure it in the required order.

Reply:
I consider this to be a transport layer feature, thus the ordering. I wanted to expose as less as possible as configuration to the end-user, and relax it if we have particular needs later. I can't think of any reason for users to interact with the compressed payload, if you do (//cc @Scottmitch) then happy to make the filters public and allow this to be configurable. WDYT?

Reply:
Use-cases I think about:
1. Metrics filters: users likely are more interested in knowing what payload size sent on the wire vs the decompressed size.
2. Metrics filter: users likely are more interested in the time it took for the HTTP exchange on the network, excluding time for post-processing.
3. Filter for checksum validation of the received payload body before decompression (checksum value may come in trailers).